src/couch_log/src/couch_log_formatter.erl

```diff
@@ -365,9 +380,29 @@ format_args([H | T], FmtAcc, ArgsAcc) ->
     {Str, _} = couch_log_trunc_io:print(H, 100),
     format_args(T, ["~s" | FmtAcc], [Str | ArgsAcc]).
 
-maybe_truncate(Fmt, Args) ->
+maybe_truncate("", [], Meta, TruncateMeta) ->
+    MaxMsgSize = couch_log_config:get(max_message_size),
+    case format_meta(Meta) of
+        "" ->
+            "";
+        MetaStr when TruncateMeta and length(MetaStr) > MaxMsgSize ->
+            %% TODO: what to do when meta formatted data is too large?
```

Comment:
If I understand the intend of the PR correctly the addition of `report` feature to logger serves a specific purpose. Which is to "report" a handful of integers along with the names of the keys.

Given the fact that we 1) do not support complex terms 2) report counters only; I think the best strategy here would be to drop the report and log an error instead. Because if it  happen it is either:

1. administrator set a `max_message_size` which is too small
2. the developer passing too much data to report call.

If we go with this route the general recommendation for developers would be to avoid using long names for counters.

Reply:
In 8f3aabe2c I've switched over to `couch_log:report` returning true or false on successful report when the message is within bounds, and I've added a metric counter `couch_log.level.report_error` to track the error on too long of messages. My rationale is that we're not going to log data about the entry that was too long to be logged, so introducing a log message won't provide additional insight over a counter.